pkg/web/counterparties.go

```diff
@@ -376,6 +390,20 @@ func (s *Server) UpdateCounterparty(c *gin.Context) {
 	// (e.g. overwrite source unknown)
 	counterparty.Source = original.Source
 
+	// ensure the website has a protocol (default to `https://`)
+	if counterparty.Website.Valid {
+		if !strings.Contains(counterparty.Website.String, "://") {
+			counterparty.Website.String = fmt.Sprintf("https://%s", counterparty.Website.String)
+		}
+		parsed, err := url.Parse(counterparty.Website.String)
+		if err != nil {
+			c.Error(err)
+			c.JSON(http.StatusInternalServerError, api.Error(err))
+			return
+		}
+		counterparty.Website.String = parsed.String()
+	}
```

Comment:
Let's force the user to provide a correct scheme/protocol rather than to guess one ourselves.

```suggestion
	if counterparty.Website.String, _ = counterparty.NormalizedWebsite(); err != nil {
		if !errors.Is(errors.ErrNullString) {
			c.Error(err)
			c.JSON(http.StatusBadRequest, api.Error("invalid website url")
			return
		}
	}
```

Reply:
I guess I missed replacing this one!